app/views/MainView/MainView.js

```diff
@@ -144,7 +145,8 @@ export default class MainView extends Component {
                 const withinAvailRides = this.withinMyAvail(
                   rides.filter(ride => ride.status === 'approved')
                 );
-
+                const outsideAvailRides = approvedRides.filter(f => !withinAvailRides.includes(f.id));
+                
```

Comment:
The way this is written, you would be comparing the ID of the `approvedRide` against the whole ride object from `withinAvailRides`. I would do something like, to make sure we're comparing ID against ID:

```suggestion
                const withinAvailRideIds = withinAvailRides.map(ride => ride.id);
                const outsideAvailRides = approvedRides.filter(ride => !withinAvailRideIds.includes(ride.id));
```